use actionable code coverage

grosser opened the pull request:
@tlunter 

before: useless noise and extra build time
```
Coverage report generated for RSpec to /Users/mgrosser/Code/tools/docker-api/coverage. 283 / 804 LOC (35.2%) covered.
-> 1.0s
```

after: clear info and no runtime overhead
```
lib/docker/util.rb new uncovered lines introduced (86 current vs 85 configured)
Lines missing coverage:
lib/docker/util.rb:15
-> 0.7s
```

I'll add this to all files if you want ... just a taste for now ...

tlunter commented:
@grosser can you just make simplecov run when not on travis? I do like the output locally.

grosser commented:
If we only run it locally then errors will slowly sneak in from users that did not run all tests locally, so running it everywhere would be best.

tlunter commented:
@grosser ok, can you add this then? Thanks.

grosser commented:
all done, fixed a few 1-off missing specs  ... also added coverage tests so they fail when a new file is added without unit tests or a new test is added without coverage check

grosser commented:
@tlunter ready!

tlunter commented:
@grosser this looks good.  Can you rebase, then I'll merge?

grosser commented:
good to go!
